Applications/spire/source/time_and_sales/time_and_sales_window.cpp

```diff
@@ -62,6 +63,9 @@ time_and_sales_window::time_and_sales_window(
   m_overlay_widget->setStyleSheet(
     "background-color: rgba(245, 245, 245, 153);");
   m_overlay_widget->hide();
+  m_loading_widget = new loading_widget(this);
```

Comment:
For this implementation, display the loading widget when a new model is passed, and destroy the widget when you receive the first time and sales from that model.